packages/bitcore-wallet-service/src/lib/server.ts

```diff
@@ -6197,8 +6219,66 @@ export class WalletService implements IWalletService {
       }
     });
   }
+
+  private coinGeckoGetCredentials() {
+    if (!config.coinGecko) throw new Error('coinGecko missing credentials');
+
+    const credentials = {
+      API: config.coinGecko.api,
+    };
+
+    return credentials;
+  }
+
+  coinGeckoGetRates(req): Promise<any> {
+    return new Promise((resolve, reject) => {
+  
+      const credentials = this.coinGeckoGetCredentials();
+      const chain = req.params?.['chain'];
+      const cacheKey = `coinGeckoRates:${chain}`;
```

Comment:
This cacheKey is incomplete since the API results depend on contractAddresses, and altCurrencies. But if we're wanting to maximize API usage, can we instead always call the API for a static set of contractAddresses and altCurrencies and then prune the cache retval to fit the query params?

Reply:
Hey, thanks for reviewing. That's not possible since `contract_addresses` is required and could be any token address that we support in the app. I removed the cache for this request ( this request is only for users that have tokens that are not supported by bitpay invoices )